### ISSUE TITLE
Refactor of taking container creation pieces from server to internal/factory

### ISSUE THREAD

**Issue by saschagrunert** [cri-o/cri-o#7275]
@haircommander can provide more input about this refactoring item. Feel free to edit the description of this roadmap topic.

**Comment by haircommander** [comment]
similar to https://github.com/cri-o/cri-o/pull/4868, we would like more of the container and sandbox container creation to be moved to the respective factory package and unit tested. Ideally this would also involve less code duplication

**Comment by hasan4791** [comment]
Have skimmed through the code & also the reference PR and here is what I've decided to do. I'm planning to move the code parts which are related to namespaces like mount/process/nw/..etc to `internal/factory/container`. Also here we would create files with "_OS" to support other platforms as well. WDYT? @haircommander @saschagrunert

**Comment by haircommander** [comment]
yup that sounds good @hasan4791 . I would say a goal is to be able to remove the call to c.Spec() in server/container_create_linux.go and have all of the spec generation operations go through the container  factory object

**Comment by hasan4791** [comment]
@dfr Can you take a look at the above approach inorder to avoid rework of segregating platform specific stuffs.

**Comment by dfr** [comment]
Looking at #4645 and comparing my container_create_freebsd.go with container_create_linux.go, it seems clear that at least for mounts, we will need some parts to be linux only - anything involving sysfs or cgroupfs for instance. For mounting files and directories from the host into the container, I use nullfs on FreeBSD - to improve sharing for this kind of thing in buildah and podman I used platform-specific types to avoid using "bind" in the code, e.g. https://github.com/containers/podman/blob/main/libpod/define/mount_freebsd.go

**Comment by hasan4791** [comment]
Unit test cases needs to be moved. WIP

**Comment by hasan4791** [comment]
/remove-lifecycle stale
Working on UTs as we need to write test cases for all the mount related methods
